src/identify/index.js

```diff
@@ -78,8 +78,8 @@ class IdentifyService {
     }
 
     // When a new connection happens, trigger identify
-    this.connectionManager.on('peer:connect', (connection) => {
-      this.identify(connection).catch(log.error)
+    this.connectionManager.on('peer:connect', async (connection) => {
+      await this.identify(connection).catch(log.error)
```

Comment:
EventEmitter handlers aren't allow to be async [according to the types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v16/events.d.ts#L330).